src/Microsoft.Azure.WebJobs.Host/Tables/TableAttributeBindingProvider.cs

```diff
@@ -49,6 +52,143 @@ public TableAttributeBindingProvider(INameResolver nameResolver, IStorageAccount
+                bindAsyncCollector); 
+
+            var bindingProvider = new GenericCompositeBindingProvider<TableAttribute>(
+                new IBindingProvider[] { bindToJArray, bindToJobject, bindAsyncCollector, original });
+
+            return bindingProvider;
+        }
+        
+        private async Task<JObject> BuildJObject(TableAttribute attribute)
+        {
+            IStorageTable table = GetTable(attribute);
+
+            IStorageTableOperation retrieve = table.CreateRetrieveOperation<DynamicTableEntity>(
+              attribute.PartitionKey, attribute.RowKey);
+            TableResult result = await table.ExecuteAsync(retrieve, CancellationToken.None);
+            DynamicTableEntity entity = (DynamicTableEntity)result.Result;
+            if (entity == null)
+            {
+                return null;
+            }
+            else
+            {
+                var obj = ConvertEntityToJObject(entity);
+                return obj;
+            }
+        }
+
+        // Build a JArray.
+        // Used as an alternative to binding to IQueryable.
+        private async Task<JArray> BuildJArray(TableAttribute attribute)
+        {
+            var table = GetTable(attribute).SdkObject;
+
+            string finalQuery = attribute.Filter;
+            if (!string.IsNullOrEmpty(attribute.PartitionKey))
+            {
+                var partitionKeyPredicate = TableQuery.GenerateFilterCondition("PartitionKey", QueryComparisons.Equal, attribute.PartitionKey);
+                if (!string.IsNullOrEmpty(attribute.Filter))
+                {
+                    finalQuery = TableQuery.CombineFilters(attribute.Filter, TableOperators.And, partitionKeyPredicate);
+                }
+                else
+                {
+                    finalQuery = partitionKeyPredicate;
+                }
+            }
+
+            TableQuery tableQuery = new TableQuery
+            {
+                TakeCount = attribute.Take, // Batch size for query. 
+                FilterString = finalQuery
+            };
+            int countRemaining = attribute.Take;
+
+            JArray entityArray = new JArray();
+            TableContinuationToken token = null;
+
+            do
+            {
+                var segment = await table.ExecuteQuerySegmentedAsync(tableQuery, token);
+                var entities = segment.Results;
+
+                token = segment.ContinuationToken;
+
+                foreach (var entity in entities)
+                {
+                    countRemaining--;
+                    entityArray.Add(ConvertEntityToJObject(entity));
+
+                    if (countRemaining == 0)
+                    {
+                        token = null;
+                        break;
+                    }
+                }                
+            }
+            while (token != null);                      
+
+            return entityArray;
+        }
```

Comment:
From script.  (See Script counterpart  PR at [https://github.com/Azure/azure-webjobs-sdk-script/pull/858](https://github.com/Azure/azure-webjobs-sdk-script/pull/858) ) 

---
In reply to: [85971905](https://github.com/Azure/azure-webjobs-sdk/pull/891#discussion_r85971905) [](ancestors = 85971905)